inc/auth.class.php

```diff
@@ -350,41 +340,32 @@ function connection_db($name, $password) {
          return false;
       }
 
-      $query = "SELECT `id`,`password`
-                FROM `glpi_users`
-                WHERE `name` = '" . $name . "'";
-      $result = $DB->query($query);
-
-      if (!$result) {
-         $this->addToError(__('Incorrect username or password'));
-         return false;
-      }
-      if ($result) {
-         if ($DB->numrows($result) == 1) {
-            $password_db = $DB->result($result, 0, "password");
-
-            if (self::checkPassword($password, $password_db)) {
-
-               // Update password if needed
-               if (self::needRehash($password_db)) {
-                  $input['id']        = $DB->result($result, 0, "id");
-                  // Set glpiID to allow passwod update
-                  $_SESSION['glpiID'] = $input['id'];
-                  $input['password']  = $password;
-                  $input['password2'] = $password;
-                  $user               = new User();
-                  $user->update($input);
-               }
-               return true;
+      // SQL query
+      $result = $DB->request('glpi_users', ['FIELDS' => ['id', 'password'], 'name' => $name]);
+
+      // Have we a result ?
+      if ($result->numrows() == 1) {
+         $maRow = $result->nextOOP();
```

Comment:
Since other variables names does not seems to use camelCase, this should not be used here. Also, variables names must be in english (os use `$row` or `$my_row` ;)